docs/specifications/tests/DNSSEC-TP/dnssec15.md

```diff
@@ -0,0 +1,188 @@
+    [Method4] and [Method5] ("NS IP").
+
+5.  Create the following empty sets:
+    1. Name server IP address and associated CDS RRset ("CDS RRsets"). A
+       name server IP can hold an empty RRset.
+    2. Name server IP address and associated CDNSKEY RRset ("CDNSKEY RRsets").
+       A name server IP can hold an empty RRset.
+    3. Name server IP address set ("Mismatch CDS/CDNSKEY").
+    4. Name server IP address set ("Has CDS No CDNSKEY").
+    5. Name server IP address set ("Has CDNSKEY No CDS").
+    5. Name server IP address set ("Has CDS And CDNSKEY").
+
+6.  Repeat the following steps for each name server IP address in *NS IP*:
+
+    1. Send the CDS query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server IP.
+       2. Else, if AA bit is not set in the DNS response, then go to next 
+          name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then go to
+          next name server IP.
+       4. Else, if the DNS response contains no CDS record in the
+          answer section, then add the name server IP and an empty RRset to
+          the *CDS RRsets* set.
+       5. Else, add the name server IP and the CDS RRset from the answer
+          section to the *CDS RRsets* set.
+    2. Send the CDNSKEY query over UDP to the name server IP address.
+       1. If no DNS response is returned, then go to next name server IP.
+       2. Else, if AA bit is not set in the DNS response, then go to next
+          name server IP.
+       3. Else, if the RCODE in the DNS response is not *NOERROR*, then go to
+          next name server IP.
+       4. Else, if the DNS response contains no CDNSKEY record in the
+          answer section, then add the name server IP and an empty RRset to
+          the *CDNSKEY RRsets* set.
+       5. Else, add the name server IP and the CDNSKEY RRset from the answer
+          section to the *CDNSKEY RRsets* set.
+    3. Go to next name server IP.
+
+7.  If the *CDS RRsets* set and the *CDNSKEY RRsets* set are empty
+    then output *[DS15_NO_CDS_CDNSKEY]* and terminate this
+    test case.
+
+8.  Repeat the following steps for each name server IP address in *NS IP*:
+
+    1. If the name server IP address has a non-empty RRset in the
+       *CDS RRsets* set, but an empty RRset in the *CDNSKEY RRsets*
+       set, then add the name server IP address to *Has CDS No CDNSKEY*.
+    2. If the name server IP address has a non-empty RRset in the
+       *CDNSKEY RRsets* set, but an empty RRset in the *CDS RRsets*
+       set, then add the name server IP address to *Has CDNSKEY No CDS*.
+    3. If the name server IP address has a non-empty RRset in both
+       sets, *CDNSKEY RRsets* and *CDS RRsets*, then add the name
+       server IP address to *Has CDS And CDNSKEY*.
+    4. Go to next name server IP.
+
+8.  If the *Has CDS No CDNSKEY* set is non-empty then output
```

Comment:
`8` -> `9`

There are some slips in the numbering below too.

Reply:
Will fix.